huaweicloud/services/rds/resource_huaweicloud_rds_sqlserver_database_privilege.go

```diff
@@ -0,0 +1,401 @@
+	}
+
+	if err = createSQLServerDatabasePrivilege(ctx, d, schema.TimeoutCreate, createSQLServerDatabasePrivilegeClient,
+		d.Get("users").(*schema.Set).List()); err != nil {
+		return diag.FromErr(err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	dbName := d.Get("db_name").(string)
+	d.SetId(instanceId + "/" + dbName)
+
+	return resourceSQLServerDatabasePrivilegeRead(ctx, d, meta)
+}
+
+func resourceSQLServerDatabasePrivilegeRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getSQLServerDatabasePrivilege: query RDS SQLServer database privilege
+	var (
+		getSQLServerDatabasePrivilegeHttpUrl = "v3/{project_id}/instances/{instance_id}/database/db_user"
+		getSQLServerDatabasePrivilegeProduct = "rds"
+	)
+	getSQLServerDatabasePrivilegeClient, err := cfg.NewServiceClient(getSQLServerDatabasePrivilegeProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	// Split instance_id and database from resource id
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return diag.Errorf("invalid ID format, must be <instance_id>/<db_name>")
+	}
+	instanceId := parts[0]
+	dbName := parts[1]
+
+	getSQLServerDatabasePrivilegePath := getSQLServerDatabasePrivilegeClient.Endpoint + getSQLServerDatabasePrivilegeHttpUrl
+	getSQLServerDatabasePrivilegePath = strings.ReplaceAll(getSQLServerDatabasePrivilegePath, "{project_id}",
+		getSQLServerDatabasePrivilegeClient.ProjectID)
+	getSQLServerDatabasePrivilegePath = strings.ReplaceAll(getSQLServerDatabasePrivilegePath, "{instance_id}", instanceId)
+
+	getSQLServerDatabasePrivilegeQueryParams := buildGetSQLServerDatabasePrivilegeQueryParams(dbName)
+	getSQLServerDatabasePrivilegePath += getSQLServerDatabasePrivilegeQueryParams
+
+	getSQLServerDatabasePrivilegeResp, err := pagination.ListAllItems(
+		getSQLServerDatabasePrivilegeClient,
+		"page",
+		getSQLServerDatabasePrivilegePath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS SQLServer database privilege")
+	}
+
+	getSQLServerDatabasePrivilegeRespJson, err := json.Marshal(getSQLServerDatabasePrivilegeResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var getSQLServerDatabasePrivilegeRespBody interface{}
+	err = json.Unmarshal(getSQLServerDatabasePrivilegeRespJson, &getSQLServerDatabasePrivilegeRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	users := utils.PathSearch("users[?name != 'rdsuser']", getSQLServerDatabasePrivilegeRespBody,
+		make([]interface{}, 0)).([]interface{})
+	if len(users) == 0 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", instanceId),
+		d.Set("db_name", dbName),
+		// 'rdsuser' is system account, it can not be managed
```

Comment:
1. it's better to add this comment at line 171
2. can we add this limitation on docs.

Reply:
done